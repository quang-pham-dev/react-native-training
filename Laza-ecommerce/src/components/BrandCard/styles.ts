import { StyleSheet } from 'react-native';

// Themes
import Colors from 'themes/Colors';
import Metrics from 'themes/Metrics';

const styles = StyleSheet.create({
  brandCardContainer: {
    justifyContent: 'center',
  },
  brandCardWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
    borderRadius: 10,
    height: 50,
    backgroundColor: Colors.secondaryBackground,
    marginRight: Metrics.margin.sm,
    paddingRight: Metrics.padding.sm,
  },
  brandLogoWrapper: {
    width: 40,
    height: 40,
    padding: Metrics.padding.xs,
    borderRadius: 10,
    marginLeft: Metrics.margin.xsm,
    marginRight: Metrics.margin.sm,
    marginVertical: Metrics.margin.xsm,
    backgroundColor: Colors.primaryBackground,
  },
  brandLogo: {
    width: '100%',
    height: '100%',
    resizeMode: 'contain',
  },
});

export default styles;
