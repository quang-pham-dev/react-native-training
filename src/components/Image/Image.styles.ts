import styled from 'styled-components/native'

// Services
import {convertToPx} from '@services/format'

interface ImageProps {
  alignSelf?: string
  mTop?: number
  mBottom?: number
  mLeft?: number
  mRight?: number
  pRight?: number
  pLeft?: number
  height: number
  width: number
  resizeMode?: string
  transform?: string
}

const ImageStyled = styled.Image<ImageProps>`
  width: ${props => convertToPx(props.width) || '100%'};
  height: ${props => convertToPx(props.height) || '100%'};
  ${props => props.alignSelf && `align-self: ${props.alignSelf}`};
  ${props => props.mTop && `margin-top: ${props.mTop}px`};
  ${props => props.mBottom && `margin-bottom: ${props.mBottom}px`};
  ${props => props.mLeft && `margin-left: ${props.mLeft}px`};
  ${props => props.mRight && `margin-right: ${props.mRight}px`};
  ${props => props.pRight && `padding-right: ${props.pRight}px`};
  ${props => props.pLeft && `padding-left: ${props.pLeft}px`};
  ${props => props.resizeMode && `resize-mode: ${props.resizeMode}`};
  ${props => props.transform && `transform: ${props.transform}`};
`

interface ImageData {
  Normal: typeof ImageStyled
}

export const Image: ImageData = {
  Normal: ImageStyled,
}
