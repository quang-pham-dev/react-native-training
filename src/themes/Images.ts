// Icons
const Icons = {
  back: require('@assets/icons/back.webp'),
  facebook: require('@assets/icons/facebook.webp'),
  twitter: require('@assets/icons/twitter.webp'),
  google: require('@assets/icons/google.webp'),
  menu: require('@assets/icons/menu.webp'),
  menuOpen: require('@assets/icons/menu-open.webp'),
  cart: require('@assets/icons/cart.webp'),
  search: require('@assets/icons/search.webp'),
  voice: require('@assets/icons/voice.webp'),
  home: require('@assets/icons/home.webp'),
  heart: require('@assets/icons/heart.webp'),
  heartLiked: require('@assets/icons/heart-liked.webp'),
  bag: require('@assets/icons/bag.webp'),
  wallet: require('@assets/icons/wallet.webp'),
  sun: require('@assets/icons/sun.webp'),
  info: require('@assets/icons/info.webp'),
  logout: require('@assets/icons/logout.webp'),
  badge: require('@assets/icons/badge.webp'),
  bagDrawer: require('@assets/icons/bag-drawer.webp'),
  heartDrawer: require('@assets/icons/heart-drawer.webp'),
  walletDrawer: require('@assets/icons/wallet-drawer.webp'),
  clock: require('@assets/icons/clock.webp'),
  ratingStart: require('@assets/icons/start-rate.webp'),
}

// Images
const Images = {
  splash: require('@assets/images/splash.png'),
}

export {Icons, Images}
