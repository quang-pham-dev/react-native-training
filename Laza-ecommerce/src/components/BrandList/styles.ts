import { StyleSheet } from 'react-native';
import { Metrics } from 'styles/themes';
const styles = StyleSheet.create({
  container: {},
  brandTitle: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingBottom: Metrics.padding.xm,
  },
});

export default styles;
