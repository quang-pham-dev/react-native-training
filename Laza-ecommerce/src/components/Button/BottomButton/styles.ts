import { StyleSheet } from 'react-native';
// Styles
import { Metrics, Colors, Fonts } from '@styles/themes/index';

const styles = StyleSheet.create({
  button: {
    paddingVertical: Metrics.padding.md,
    paddingHorizontal: Metrics.padding.xl,
    alignItems: 'center',
  },
  title: {
    color: Colors.white,
    textTransform: 'none',
    fontFamily: Fonts.fontFamily.Inter_600SemiBold,
    fontSize: Fonts.size.sm,
    lineHeight: Fonts.lineHeight.lg,
  },
});

export default styles;
