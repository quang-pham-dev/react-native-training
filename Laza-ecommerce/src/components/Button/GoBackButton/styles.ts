// Styles
import { Metrics } from '@styles/themes/index';
import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  button: {
    marginTop: Metrics.margin.customXxl,
    marginLeft: Metrics.margin.lg,
    alignItems: 'center',
  },
  icon: {
    width: Metrics.icons.semiLarge,
    height: Metrics.icons.semiLarge,
  },
});

export default styles;
