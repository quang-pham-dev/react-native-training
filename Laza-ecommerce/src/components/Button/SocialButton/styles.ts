import { StyleSheet } from 'react-native';
// Styles
import { Metrics, Colors, Fonts } from '@styles/themes/index';

const styles = StyleSheet.create({
  button: {
    paddingVertical: Metrics.padding.md,
    paddingHorizontal: Metrics.padding.xl,
    alignItems: 'center',
    borderColor: Colors.transparent,
    borderWidth: 1,
    backgroundColor: Colors.white,
  },
  title: {
    color: Colors.white,
    textTransform: 'none',
    fontFamily: Fonts.fontFamily.Inter_600SemiBold,
    fontSize: Fonts.size.sm,
    lineHeight: Metrics.height.custom,
  },
  icon: {
    width: Metrics.icons.custom,
    height: Metrics.icons.custom,
  },
});

export default styles;
