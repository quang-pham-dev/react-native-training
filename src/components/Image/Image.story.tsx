import * as React from 'react'

import {storiesOf} from '@storybook/react-native'
import {select} from '@storybook/addon-knobs'
import {StoryScreen, Story, UseCase} from '../../../storybook/views'

// Styled
import {Image} from './Image.styles'

// Themes
import {Images} from '@themes/Images'

declare let module: NodeModule

const images = {...Images}

storiesOf('Image', module)
  .addDecorator(fn => <StoryScreen>{fn()}</StoryScreen>)
  .add('Normal', () => {
    const image = select('source', Object.keys(images), 'splash')

    return (
      <Story>
        <UseCase text="Image normal">
          <Image.Normal
            source={images[image as keyof typeof Images]}
            height={230}
          />
        </UseCase>
      </Story>
    )
  })
