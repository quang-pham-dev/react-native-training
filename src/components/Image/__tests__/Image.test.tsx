import React from 'react'
import {render} from '@testing-library/react-native'

// Styled
import {Image} from '../Image.styles'

import {Images} from '@themes/Images'

const props = {
  source: Images.splash,
  height: 300,
}

describe('Image', () => {
  const normalImage = render(<Image.Normal {...props} />)
  test('should render normal image correctly', () => {
    const tree = normalImage.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
