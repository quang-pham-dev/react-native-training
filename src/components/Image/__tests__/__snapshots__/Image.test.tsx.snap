// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render normal image correctly 1`] = `
<Image
  height={300}
  source={
    Object {
      "testUri": "../../../src/assets/images/splash.png",
    }
  }
  style={
    Array [
      Object {
        "height": 300,
        "width": "100%",
      },
    ]
  }
/>
`;
